geode-core/src/test/java/org/apache/geode/admin/internal/FinishBackupRequestTest.java

```diff
@@ -0,0 +1,213 @@
+  private File baselineDir;
+
+  private InternalDistributedMember localMember;
+  private InternalDistributedMember member1;
+  private InternalDistributedMember member2;
+
+  private Set<InternalDistributedMember> recipients;
+
+  @Before
+  public void setUp() throws Exception {
+    // mocks here
+    replyProcessor = mock(FinishBackupReplyProcessor.class);
+    dm = mock(DM.class);
+    cache = mock(InternalCache.class);
+    backupManager = mock(BackupManager.class);
+    targetDir = mock(File.class);
+    baselineDir = mock(File.class);
+
+    when(dm.getCache()).thenReturn(cache);
+    when(dm.getDistributionManagerId()).thenReturn(localMember);
+    when(cache.getBackupManager()).thenReturn(backupManager);
+    when(replyProcessor.getResults()).thenReturn(Collections.emptyMap());
+
+    localMember = mock(InternalDistributedMember.class);
+    member1 = mock(InternalDistributedMember.class);
+    member2 = mock(InternalDistributedMember.class);
+
+    recipients = new HashSet<>();
+    recipients.add(member1);
+    recipients.add(member2);
+
+    finishBackupRequest =
+        new FinishBackupRequest(dm, recipients, replyProcessor, targetDir, baselineDir, false);
+  }
+
+  @Test
+  public void getRecipientsReturnsRecipientMembers() throws Exception {
+    assertThat(finishBackupRequest.getRecipients()).hasSize(2).contains(member1, member2);
+  }
+
+  @Test
+  public void getRecipientsDoesNotIncludeNull() throws Exception {
+    InternalDistributedMember nullMember = null;
+
+    assertThat(finishBackupRequest.getRecipients()).doesNotContain(nullMember);
+  }
+
+  @Test
+  public void sendShouldUseDMToSendMessage() throws Exception {
+    finishBackupRequest.send();
+
+    verify(dm, times(1)).putOutgoing(finishBackupRequest);
+  }
+
+  @Test
+  public void sendShouldWaitForRepliesFromRecipients() throws Exception {
+    finishBackupRequest.send();
+
+    verify(replyProcessor, times(1)).waitForReplies();
+  }
+
+  @Test
+  public void sendShouldReturnResultsContainingRecipientsAndLocalMember() throws Exception {
+    Set<PersistentID> localMember_PersistentIdSet = new HashSet<>();
+    localMember_PersistentIdSet.add(mock(PersistentID.class));
+    Set<PersistentID> member1_PersistentIdSet = new HashSet<>();
+    member1_PersistentIdSet.add(mock(PersistentID.class));
+    Set<PersistentID> member2_PersistentIdSet = new HashSet<>();
+    member2_PersistentIdSet.add(mock(PersistentID.class));
+    member2_PersistentIdSet.add(mock(PersistentID.class));
+    Map<DistributedMember, Set<PersistentID>> expectedResults = new HashMap<>();
+    expectedResults.put(localMember, localMember_PersistentIdSet);
+    expectedResults.put(member1, member1_PersistentIdSet);
+    expectedResults.put(member2, member2_PersistentIdSet);
+    when(replyProcessor.getResults()).thenReturn(expectedResults);
+
+    Map<DistributedMember, Set<PersistentID>> results = finishBackupRequest.send();
+
+    assertThat(results).isEqualTo(expectedResults);
+  }
+
```

Comment:
Above comments apply to below test in this class and to similar tests in other classes.